haproxy-healthcheck/check.py

```diff
@@ -1,80 +1,93 @@
+#!/usr/bin/env python3
+import asyncio
 import os
-import socket, requests
+from typing import List
 
-from _thread import *
-import threading
+import httpx
 
-print_lock = threading.Lock()
+from logger import log
 
-couchdb_servers = os.environ['COUCHDB_SERVERS'].split(',')
+couchdb_servers: List[str] = os.environ["COUCHDB_SERVERS"].split(",")
 # Example: COUCHDB_SERVERS="couchdb.1,couchdb.2,couchdb.3"
-username = os.environ['COUCHDB_USER']
-password = os.environ['COUCHDB_PASSWORD']
+username: str = os.environ["COUCHDB_USER"]
+password: str = os.environ["COUCHDB_PASSWORD"]
 
-def get_membership_endpoint(couchdb_url):
-    return f'http://{username}:{password}@{couchdb_url}:5984/_membership'
 
-def close_connection(connection):
-    print('Closing connection ...')
-    connection.close()
+def get_membership_endpoint(couchdb_url: str) -> str:
+    return f"http://{username}:{password}@{couchdb_url}:5984/_membership"
 
-def send_down_response(conn):
-    print('Sending response: Down (CouchDB cluster not okay) ...')
-    conn.send(b'down\n')
-    close_connection(conn)
 
-def send_up_response(conn):
-    print('Sending response: Up (Couchdb cluster okay) ...')
-    conn.send(b'up\n')
-    close_connection(conn)
+def _send_down_response(writer: asyncio.StreamWriter) -> None:
+    _send_and_close(writer, b"down\n")
 
-def threaded(conn):
+
+def _send_up_response(writer: asyncio.StreamWriter) -> None:
+    _send_and_close(writer, b"up\n")
+
+
+def _send_and_close(writer: asyncio.StreamWriter, message: bytes) -> None:
+    log.info("Response: %r", message)
```

Comment:
What is our default log level? I wonder if this becomes too noisy. If we have 3 servers, each making a check every 5 seconds, this log file will be gigantic with 3 lines for every request. Can we maybe compress this to a single line? 
In order for our containers to get environment variables, we need to add them to the compose file templates and to the upgrade service (for docker deployments). is `APP_LOGLEVEL` the new variable that controls the log level?

Reply:
Perfect!